### PR TITLE
[TIMOB-23691] iOS: Expose statusBarBackgroundColor

### PR DESCRIPTION
**Pull request by kopiro**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-23691

### CODE REVIEW

File: iphone/Classes/TiUIWindowProxy.m [hunk 1]
```diff
@@ -363,6 +363,23 @@ -(void)setBarColor:(id)colorString
...
+    if (controller != nil && [controller navigationController] != nil)
+    {
+        if (statusbarBg == nil) {
+            statusbarBg = [[UIView alloc] initWithFrame:[UIApplication sharedApplication].statusBarFrame];
+            [[controller navigationController].view addSubview:statusbarBg];
+        }
+        TiColor* newColor = [TiUtils colorValue:color];
+        statusbarBg.backgroundColor = [newColor _color];
+    }
+}
```
Comment (hansemannn) [inline iphone/Classes/TiUIWindowProxy.m]: Not sure if that's the best way to go. I like the style proposed here: http://stackoverflow.com/a/34658477/5537752

Also, `statusbarBg` should be released from memory every time it is set again, as well as in the `dealloc` method to clean it from memory after the gets GC'd.

Please try the above-mentioned way to do it.

**Comment by kopiro** [comment]
@hansemannn Wow, thank you!
It works like a charm.

Pushed ;)

**Comment by hansemannn** [comment]
Happy to hear. One thing that concerns me is, that we can set the `statusBarBackgroundColor` in every `Ti.UI.Window` context but it is set globally. So either we place it in `Ti.UI` instead or find a way to make it window-depending.

**Comment by hansemannn** [comment]
And remove `NSString *color = [TiUtils stringValue:colorString];` + replace it with `ENSURE_SINGLE_ARG(value, NSString)`. Name it `value` as well.

**Comment by kopiro** [comment]
Mmm, theoretically we can just call the `setBackground` on every Window open.

**Comment by hansemannn** [comment]
Right. Then please move it to `Ti.UI` and let the user decide how to handle it. Guess that keeps it the most flexible for both use-cases (global- and context-based).

**Comment by kopiro** [comment]
Hi @hansemannn, I thought about, could be better to move to `Ti.UI.iOS` instead?

**Comment by hansemannn** [comment]
Excellent idea! Now please restore the TiUIWindowProxy.m with the original one and place the docs in `apidoc/Titanium/UI/iOS/iOS.yml` (should be the one).

**Comment by hansemannn** [comment]
Can you please pull from master? Cloning your PR locally includes old stuff like the manual scons-scripts.

**Comment by hansemannn** [comment]
Approved!

**Pull request merged**